universal-application-tool-0.0.1/app/repository/ProgramRepository.java

```diff
@@ -121,8 +122,14 @@ public Program createOrUpdateDraft(Program existingProgram) {
             program.getSlug();
             program.save();
           }
+          ImmutableList<Program> activePrograms =
+              versionRepository.get().getActiveVersion().getPrograms();
+          List<Program> programsMatchingSlug =
+              ebeanServer.find(Program.class).where().eq("slug", slug).findList();
```

Comment:
(question): in the Program model it says there's a many to many relationship between versions. Doesn't each program just get one version?

I'm wondering if it'd be more effective to use the joined table instead of streaming through all active programs.

Reply:
Each program has many versions, yep - if a program is unchanged, the same program id will be in two different versions.  There is only one active and one draft version at a time, though.

Reply:
It might be more efficient but it's also harder to write in ebean's query DDL - and we're talking about a database query returning a few dozen rows.  At this scale we can just do the easy thing.